deployment/aws-python-storage-stack-automation/storage_stack.yaml

```diff
@@ -0,0 +1,355 @@
+              cft_client = boto3.client("cloudformation")
+              print("Creating stack ..")
+              cft_client.create_stack(
+                  StackName="C1-FSS-Storage-" + bucket_name,
+                  TemplateURL="https://file-storage-security.s3.amazonaws.com/latest/templates/FSS-Storage-Stack.template",
+                  Parameters=[
+                      ExternalID,
+                      S3BucketToScan,
+                      scanner_queue_url,
+                      Trigger_with_event,
+                      scanner_aws_account,
+                      S3_Encryption,
+                  ],
+                  Capabilities=["CAPABILITY_IAM"],
+              )
+              cft_waiter = cft_client.get_waiter("stack_create_complete")
+              cft_waiter.wait(StackName="C1-FSS-Storage-" + bucket_name)
+              res = cft_client.describe_stacks(StackName="C1-FSS-Storage-" + bucket_name)
+              storage_stack = res["Stacks"][0]["Outputs"][2]["OutputValue"]
+              add_to_cloudone(ws_key, stack_id, storage_stack)
+          # register storage stack to cloud one
+          def add_to_cloudone(ws_key, stack_id, storage_stack):
+              print("FSS StorageRole Arn: " + storage_stack)
+              # add to c1
+              payload = {
+                  "type": "storage",
+                  "scannerStack": stack_id,
+                  "provider": "aws",
+                  "details": {"managementRole": storage_stack},
+              }
+              encoded_msg = json.dumps(payload)
+              resp = http.request(
+                  "POST",
+                  stacks_api_url+"stacks",
+                  headers={
+                      "Content-Type": "application/json",
+                      "api-secret-key": ws_key,
+                      "Api-Version": "v1",
+                  },
+                  body=encoded_msg,
+              )
+              transform = json.loads(resp.data.decode("utf-8"))
+              url = "https://cloudone.trendmicro.com/api/filestorage/stacks/"+transform['stackID']
+              try:
+                check_status(ws_key, url)
+              except json.decoder.JSONDecodeError:
+                time.sleep(1)
+                check_status(ws_key, url)
+          #check storage stack status
+          def check_status(ws_key, url):
+              #gather stack status
+              st_call = http.request('GET', url , headers = {'api-secret-key': ws_key, 'Api-Version': 'v1'})
+              x = json.loads(st_call.data.decode('utf-8'))['status']
+              print("Status: " + x)
+              while x == 'creating':
+                  st_call = http.request('GET', url , headers = {'api-secret-key': ws_key, 'Api-Version': 'v1'})
+                  x = json.loads(st_call.data.decode('utf-8'))['status']
+              else:
+                  print("Status: " + x)
+                  print('Deployed Successfully')
```

Comment:
It could also be other status like `failed`. Explicitly checking whether it's 'ok' is more accurate.